include/drivers/i2s.h

```diff
@@ -373,6 +374,32 @@ static inline struct i2s_config *i2s_config_get(struct device *dev,
 	return api->config_get(dev, dir);
 }
 
+/**
+ * @brief Get I2S controller state.
+ *
+ * The dir parameter specifies if the state of the Transmit (TX) or
+ * Receive (RX) direction will be returned.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param dir Stream direction: RX or TX as defined by I2S_DIR_*.
+ *
+ * @note The state may change independently of API calls to the i2s driver.
+ *
+ * @retval -ENOTSUP If the underlying driver does not support i2s_state_get.
+ * @retval state A positive state value if the state was returned successfully.
```

Comment:
This is still not correct.  `-ENOTSUP` is not compatible with `enum i2s_state` as a return value.  The return value must be changed to `int`.

Also `I2S_STATE_NOT_READY` has integer value zero, so the state retval should be "A non-negative state value..."

Reply:
Thank you, I missed that. Corrections have been pushed.